docs/source/peer_ledger_snapshot.md

```diff
@@ -0,0 +1,153 @@
+
+If the request is successful, you will see a `Snapshot request submitted successfully` message.
+
+You can list the pending snapshots by issuing a command similar to:
+
+```
+peer snapshot listpending -C testchannel --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+You will see a response similar to:
+
+```
+Successfully got pending snapshot requests [970]
+```
+
+When a snapshot has been generated for a particular block height, the pending request for that block height will no longer appear in the list. You can also verify that a snapshot has been create successfully by looking at the peer logs. You will see a message similar to `Generated snapshot block-<block number>`.
+
+To delete a snapshot request, simply exchange `submitrequest` with `cancelrequest`. For example:
+
+```
+peer snapshot cancelrequest -C testchannel -b 1000 --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+If you submit the `listpending` command again, the snapshot should no longer appear.
+
+### Contents of a snapshot
+
+Snapshots are stored on a peer in a snapshot folder that is further organized by channel name and the ledger height when the snapshot was taken: `{snapshotsFolder}/{channelName}/{blockHeight}`.
+
+As mentioned above, the snapshot contains four folders:
+
+* **Public state**
+  * This includes the latest state of all of the assets on the channel (for example, the current owner of a car).
+* **Private data hashes**
+  * Recall from our documentation on private data that while the actual transaction data of a private data transaction is not stored on the public ledger of the channel, hashes of the data are stored. This allows the private data to be verified against the hash, for example by an organization that is added to a private data collection.
+* **Transactions IDs**
+  * Starting with Fabric v2.0, entries in the block store index take the form of `TxID_BlkNum_TxNum`. This index can be filtered for the the transactions IDs that are added after the snapshot height by evaluating block number (`BlkNum`) portion of the index entry.
+* **Collection config history**
+  * This contains the private data collections the peer is a member of.
+
+In addition, the snapshot contains several metadata files that help the data in the snapshot to be verified:
+
+* Channel name
+* Height of the ledger at the time the snapshot was taken
+* BlockHash for the last block in the snapshot
+* Hash of the public state
+* Hash of the private data hashes
+* Hash of transaction IDs
+* Hash of collection config history
+* A representative hash of the snapshot, computed by combining all of the above
+* `BlockCommitHash` for the last block in the snapshot
+* Compression algorithm used for the data files
```

Comment:
* **_snapshot_signable_metadata.json**
<br>This snapshot metadata file is expected to be same across snapshots of a channel for a particular height. This file contains a JSON record with the following fields
  * channel_name
  * last_block_number
  * last_block_hash
  * previous_block_hash
  * state_db_type (the value of this field will be either of `CouchDB` and `SimpleKeyValueDB`. The latter will be the value if leveldb is used for statedb)
  * snapshot_files_raw_hashes (the value of this will be another JSON record that caontains the hashes of the files present for the above mentioned data items)

* **_snapshot_additional_metadata.json** 
<br>This another metadata file contains a JSON record with the following two fields
  * snapshot_hash (this is the hash of the file `_snapshot_signable_metadata.json` and can be treated as a hash of the snapshot)
  * last_block_commit_hash (if the snapshot generating peer is equipped to compute the block commit hashes)

Reply:
This is great, thank you!